pages/en/publish/testing-digital-goods-api.md

```diff
@@ -0,0 +1,32 @@
+---
+title: Testing the Digital Goods API Integration
+metadesc: How to ensure that your Digital Goods Integration is working properly.
+date: 2021-11-05
+weight: -3
+---
+
+# License testing on the Play Store
+
+When testing using the Google Play Store and using [application license testing](https://support.google.com/googleplay/android-developer/answer/6062777), subscription licenses are set to renew every five minutes rather than the time interval set by your license.
+
+# Test cases
+
+Testing the digital goods api integration can help make sure that there are no unexpected bugs for your users in production. In order to better check for these integrations, we recommend testing the following in your application:
+
+- **Upgrade and Downgrade Flow -** If you sell multiple subscriptions, you want to ensure that the right subscription is granted to your user when they upgrade and downgrade from different tiers of the subscription. You also want to make sure that the billing options such as a free trial or special pricing incentive are enabled. If you have a user that changes their subscription, check that the proration occurs correctly and that granting / removing entitlements also happens correctly.
+- **Onetime purchases and repeatable purchases are handled correctly -** One time purchases only occur once and cannot be accidentally purchased again. Repetable purchases can happen immediately after they are created. Entitlments are generated on the beackend for both.
+- **Cancel Subscriptions From App -** You want to make sure that subscriptions from your app are handled correctly. (See the section titled [Let users manage subscriptions](https://chromeos.dev/en/publish/pwa-play-billing#let-users-manage-subscriptions) for information on how to direct your users to cancel from within the app) **Note:** When a user cancels their subscription, the subscription doesn’t officially expire until the end of their current billing period. So, the user should still have access to their subscription plan features until the subscription expires.
+- **Digital Goods and other payment options are served correctly -** If a user enters a digital storefront from an app context, they should see Google Play Billing available when going to subscribe. If the user enters the payment flow from a browser context, they should see the alternative payment flows served.
+- **[Out of App Purchase](https://chromeos.dev/en/publish/pwa-play-billing#out-of-app-purchases) Flows**
+  - **Subscription from promo code -** Subscriptions come in as unacknowledged purchases. When the user subscribes and opens the app, the subscription should be verified, entitlements should be granted, and the subscription should be acknowledged.
+  - **One time purchase from promo code -** One time purchases should be acknowledged but entitlements may not be granted. Once the user logs into the app, make sure that the entitlement is granted.
```

Comment:
The wording of `One time purchases should be acknowledged but entitlements may not be granted.` is a bit vague. I could see how someone would read it and think that they need to acknowledge but its ok if they don't grant entitlement. Maybe something like `One time purchases will already be acknowledged by Play, but the user won't have the entitlement yet.`